app/src/main/java/com/termux/api/cron/CronWorker.java

```diff
@@ -166,4 +174,16 @@ private void sendKillIntent() {
     private void scheduleNextExecution() {
         CronScheduler.scheduleAlarmForJob(getApplicationContext(), jobId);
     }
+
+    private static String createAppShellName(int jobId, Uri executableUri) {
+        char[] allowedCharsArray = ("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789").toCharArray();
+        char[] randomId = new char[6];
+        Random random = new SecureRandom();
+        for (int i = 0; i < randomId.length; i++) {
+            randomId[i] = allowedCharsArray[random.nextInt(allowedCharsArray.length)];
+        }
+
+        return String.format(Locale.getDefault(),
+                "%s-%d-%s", executableUri.getLastPathSegment(), jobId, String.copyValueOf(randomId));
```

Comment:
Use `FileUtils.getFileBasename((UriUtils.getUriFilePathWithFragment(executableUri))` possibly with `null` check, read the java docs for why. However, I have locally added the `com.termux.execute.executable` extra for the path in `TermuxService` to get away from this uri mess.

Reply:
Did you actually encountered paths with # ? I know it's allowed, but it feels very wrong ;)

Anyway I applied your suggestion. Thanks.

Reply:
Yeah, some user reported it somewhere for `termux-open`, it got fixed with https://github.com/termux/termux-app/commit/3e518a6a75a26e64edd6b1f4643640c2459f5407

Linux allows all characters/bytes other than `null` byte, `#` isn't that unlikely, it's an ascii character after all.